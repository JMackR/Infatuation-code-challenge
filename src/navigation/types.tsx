import type { StackScreenProps } from '@react-navigation/stack'
import type { BottomTabScreenProps } from '@react-navigation/bottom-tabs'
import { NavigableRoute } from './routes'
import type { CompositeScreenProps, NavigatorScreenParams } from '@react-navigation/native'

export type RootStackParamList = {
	[NavigableRoute.Tabs]: NavigatorScreenParams<TabParamList>
}
export type RootStackScreenProps<T extends keyof RootStackParamList> = StackScreenProps<RootStackParamList, T>

export type TabParamList = {
	[NavigableRoute.Search]: undefined
	[NavigableRoute.Favorites]: undefined

}
export type TabScreenProps<T extends keyof TabParamList> = CompositeScreenProps<
	BottomTabScreenProps<TabParamList, T>,
	RootStackScreenProps<keyof RootStackParamList>
>
