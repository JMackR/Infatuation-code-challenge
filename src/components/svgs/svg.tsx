import React from 'react'
import { LocalSVGSource, SvgPropsBase } from './svg.types'
import { TouchableOpacity } from 'react-native'

export const SVG = (props: SvgPropsBase) => {
  const { localSVG, tint, testID } = props
  const { SVG, size } = localSVG as LocalSVGSource
  if (props.onPress) {
    return (
      <TouchableOpacity onPress={props.onPress} testID={testID}>
        <SVG {...size} fill={tint} />
      </TouchableOpacity>
    )
  } else {
    return <SVG {...size} fill={tint} />
  }

}
