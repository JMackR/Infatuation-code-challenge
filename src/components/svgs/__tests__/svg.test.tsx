import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import { SVG } from '../svg'
import { LocalSVGSource, SvgPropsBase } from '../svg.types'

const localSVGSource: LocalSVGSource = {
  SVG: () => <svg>This is a local SVG</svg>,
  size: {
    width: 30,
    height: 30,
  }
}
describe('SVG Snapshot Tests', () => {
  let props: SvgPropsBase
  beforeEach(() => {
    props = {
      onPress: jest.fn(),
    }
  })
  test('SVG renders localSVG ', () => {
    const tree = render(<SVG localSVG={localSVGSource} />)
    expect(tree).toMatchSnapshot()
  })

  test('SVG renders localSVG with overwritten width/height', () => {
    const tree = render(<SVG localSVG={{ SVG: localSVGSource.SVG, size: { width: 120, height: 120 } }} />)
    expect(tree).toMatchSnapshot()
  })

  test('SVG renders tintable localSVG with white tint', () => {
    const tree = render(<SVG localSVG={localSVGSource} tint='#ffffff' />)
    expect(tree).toMatchSnapshot()
  })

  test('SVG renders tintable localSVG with black tint', () => {
    const tree = render(<SVG localSVG={localSVGSource} tint='#000000' />)
    expect(tree).toMatchSnapshot()
  })

  test('SVG is pressable', () => {
    const { getByTestId } = render(<SVG localSVG={localSVGSource} {...props} testID={'hibob'} />)
    fireEvent.press(getByTestId('hibob'));
    expect(props.onPress).toHaveBeenCalledTimes(1)
  })

})

