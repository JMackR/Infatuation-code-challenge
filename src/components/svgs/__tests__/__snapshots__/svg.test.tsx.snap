// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVG Snapshot Tests SVG renders localSVG  1`] = `
<svg>
  This is a local SVG
</svg>
`;

exports[`SVG Snapshot Tests SVG renders localSVG with overwritten width/height 1`] = `
<svg>
  This is a local SVG
</svg>
`;

exports[`SVG Snapshot Tests SVG renders tintable localSVG with black tint 1`] = `
<svg>
  This is a local SVG
</svg>
`;

exports[`SVG Snapshot Tests SVG renders tintable localSVG with white tint 1`] = `
<svg>
  This is a local SVG
</svg>
`;
